moto/s3/responses.py

```diff
@@ -1259,8 +1259,16 @@ def _key_response_put(self, request, body, bucket_name, query, key_name, headers
             )
 
             if key is not None:
-                if key.storage_class in ["GLACIER", "DEEP_ARCHIVE"]:
+                if key.storage_class in "GLACIER":
```

Comment:
I'm not entirely how `in` works with strings (I think `x in y` is true if `x` is a substring of `y`), but I think you might want `==` here instead. 

Do you need to treat DEEP_ARCHIVE differently from GLACIER? Is restoring any different between the two?

Reply:
docs say they're restored identically

for reference, `a in b` checks that a is a member of iterable b, so `1 in [1, 2, 3]` is True, as is `"a" in "abc"`, so yes you want `==`